src/server/JasmineGraphInstanceService.cpp

```diff
@@ -80,4301 +144,5243 @@ void *instanceservicesession(void *dummyPt) {
+    std::vector<std::string> fileNames;
+    map<long, unordered_set<long>> aggregatedCentralStore;
+    std::string centralGraphIdentifier = graphId + +"_centralstore_" + partitionId;
+    std::string dataFolder = Utils::getJasmineGraphProperty("org.jasminegraph.server.instance.datafolder");
+    std::string workerCentralStoreFile = dataFolder + "/" + centralGraphIdentifier;
+    instance_logger.log("###INSTANCE### Loading Central Store : Started " + workerCentralStoreFile, "info");
+    JasmineGraphHashMapCentralStore workerCentralStore =
+        JasmineGraphInstanceService::loadCentralStore(workerCentralStoreFile);
+    instance_logger.log("###INSTANCE### Loading Central Store : Completed", "info");
+    map<long, unordered_set<long>> workerCentralGraphMap = workerCentralStore.getUnderlyingHashMap();
 
-            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
-            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::OK, "info");
+    map<long, unordered_set<long>>::iterator workerCentalGraphIterator;
 
-            bzero(data, INSTANCE_DATA_LENGTH);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string workerList = (data);
-            workerList = Utils::trim_copy(workerList, " \f\n\r\t\v");
-            instance_logger.log("Received WorkerList: " + workerList, "info");
-
-            std::vector<string> workerSockets;
-            stringstream wl(workerList);
-            string intermediate;
-            while (getline(wl, intermediate, ',')) {
-                workerSockets.push_back(intermediate);
-            }
+    for (workerCentalGraphIterator = workerCentralGraphMap.begin();
+         workerCentalGraphIterator != workerCentralGraphMap.end(); ++workerCentalGraphIterator) {
+        long startVid = workerCentalGraphIterator->first;
+        unordered_set<long> endVidSet = workerCentalGraphIterator->second;
 
-            auto t_start = std::chrono::high_resolution_clock::now();
+        unordered_set<long> aggregatedEndVidSet = aggregatedCentralStore[startVid];
+        aggregatedEndVidSet.insert(endVidSet.begin(), endVidSet.end());
+        aggregatedCentralStore[startVid] = aggregatedEndVidSet;
+    }
 
-            map<long, long> degreeDistribution =
-                calculateLocalInDegreeDist(graphID, partitionID, graphDBMapLocalStores, graphDBMapCentralStores);
+    std::vector<std::string> paritionIdList = Utils::split(partitionIdList, ',');
+    std::vector<std::string>::iterator partitionIdListIterator;
 
-            instance_logger.log("In Degree Dist size: " + to_string(degreeDistribution.size()), "info");
+    for (partitionIdListIterator = paritionIdList.begin(); partitionIdListIterator != paritionIdList.end();
+         ++partitionIdListIterator) {
+        std::string aggregatePartitionId = *partitionIdListIterator;
+        struct stat s;
```

Comment:
Its better if we can choose a better variable name than `s`. A meaningful word would be better.

Reply:
If this is used to check if the file exists, we can do it with `access(2)` function with permission checks also.

Reply:
Fixed it in the next PR